src/fd.cpp

```diff
@@ -85,21 +107,16 @@ static int wait_thread(void *arg) {
     } else if (waitFor_ms > LATER_POLL_INTERVAL) {
       waitFor_ms = LATER_POLL_INTERVAL;
     }
-    ready = LATER_POLL_FUNC(pollfds.data(), args->num_fds, static_cast<int>(waitFor_ms));
+    ready = LATER_POLL_FUNC(args->fds->data(), args->num_fds, static_cast<int>(waitFor_ms));
     if (args->flag->load()) return 1;
     if (ready) break;
   }
 
   // store pollfd revents in args->results for use by callback
+
   if (ready > 0) {
-    for (int i = 0; i < args->rfds; i++) {
-      (*args->results)[i] = pollfds[i].revents == 0 ? 0 : pollfds[i].revents & POLLIN ? 1: R_NaInt;
-    }
-    for (int i = args->rfds; i < (args->rfds + args->wfds); i++) {
-      (*args->results)[i] = pollfds[i].revents == 0 ? 0 : pollfds[i].revents & POLLOUT ? 1 : R_NaInt;
-    }
-    for (int i = args->rfds + args->wfds; i < args->num_fds; i++) {
-      (*args->results)[i] = pollfds[i].revents != 0;
+    for (int i = 0; i < args->num_fds; i++) {
+      (*args->results)[i] = (*args->fds)[i].revents == 0 ? 0 : (*args->fds)[i].revents & (POLLIN | POLLOUT) ? 1: R_NaInt;
     }
   } else if (ready == 0) {
     for (int i = 0; i < args->num_fds; i++) {
```

Comment:
I believe this isn't necessary because `std::vector<int>` zeroes itself on initialization.

Reply:
Good catch - it's zero initialized as I do so in the `ThreadArgs` constructor when I pass `num_fds`, so I've taken out in [1bafc3b](https://github.com/shikokuchuo/later/pull/1/commits/1bafc3beb9a4ad49d617a8ff229ce094dfa3d3e8).